src/server/db/actions/GameAction.ts

```diff
@@ -72,70 +72,124 @@ export async function editGame(allData: nextEditGame) {
   //Don't modify until sure of how data is used in the API
   await connectMongoDB();
   const data: IEditGame = allData.data;
-  try {
-    //Ensure every ObjectID actually represents a Document
-    //Ah yes
-    if (data && data.themes) {
-      const themePromises = data.themes.map((theme) =>
-        ThemeModel.findById(theme),
-      );
-      const themeResults = await Promise.all(themePromises);
-      themeResults.forEach((result, index) => {
-        if (!result) {
-          throw new GenericUserErrorException(
-            `ObjectID ${data.themes![index]} not a present theme.`,
-          ); //Using non-null assertion, as if condition should ensure data.tags is non-null
-        }
-      });
+  //Ensure every ObjectID actually represents a Document
+  //Ah yes
+  if (data && data.themes) {
+    const themeResults = await ThemeModel.find({ id: { $in: data.themes } });
+    if (themeResults.length !== data.themes.length) {
+      throw new GenericGameErrorException(
+        "One of the given themes does not exist.",
+      ); //Using non-null assertion, as if condition should ensure data.tags is non-null
     }
-    if (data && data.tags) {
-      const tagPromises = data.tags.map((tag) => TagModel.findById(tag));
-      const tagResults = await Promise.all(tagPromises);
-      tagResults.forEach((result, index) => {
-        if (!result) {
-          throw new GenericUserErrorException(
-            `ObjectID ${data.tags![index]} is not a  present tag.`,
-          ); //Using non-null assertion, as if condition should ensure data.tags is non-null
-        }
+  }
+  if (data && data.tags) {
+    const tagResults = await TagModel.find({ id: { $in: data.tags } });
+    if (tagResults.length !== data.tags.length) {
+      throw new GenericGameErrorException(
+        "One of the given tags does not exist.",
+      ); //Using non-null assertion, as if condition should ensure data.tags is non-null
+    }
+  }
+
+  const result = await GameModel.findByIdAndUpdate(allData.id, allData.data, {
+    new: false,
+  });
+  if (!result) {
+    throw new GenericGameErrorException("Game with given ID does not exist.");
+  }
+  return result;
+}
+export async function getSelectedGames(
+  query: z.infer<typeof GetGameQuerySchema>,
+) {
+  await connectMongoDB();
+  //Want to be able to filter by specific tags,game_builds, etc.
+  //Find the corresponding id to the Theme tag.
+  let filters: FilterQuery<IGame> = {}; //There should be a better way to build the query gradually
+  if (query.theme) {
+    //Find corresponding themeObject in mongodb
+    const found_theme: (ITheme & { id: ObjectId }) | null =
```

Comment:
You shouldn't need to assert this type

Nit: use camelCase

Reply:
Anytime you reference ObjectId as a **type**, use `Types.ObjectId` instead of just `ObjectId`. Read [here](https://mongoosejs.com/docs/schematypes.html#:~:text=A%20SchemaType%20is%20different%20from%20a%20type.%20In%20other%20words%2C%20mongoose.ObjectId%20!%3D%3D%20mongoose.Types.ObjectId)